sdk/storage/storage-blob/test/blobclient.spec.ts

```diff
@@ -230,42 +230,86 @@ describe("BlobClient", () => {
   });
 
   it("undelete", async () => {
-    const properties = await blobServiceClient.getProperties();
+    let properties = await blobServiceClient.getProperties();
     if (!properties.deleteRetentionPolicy!.enabled) {
       await blobServiceClient.setProperties({
         deleteRetentionPolicy: {
           days: 7,
           enabled: true
         }
       });
-      await delay(15 * 1000);
+      // await delay(15 * 1000);
+      properties = await blobServiceClient.getProperties();
+      assert.ok(properties.deleteRetentionPolicy!.enabled, "deleteRetentionPolicy should be enabled.");
```

Comment:
Server doesn't guarantee soft delete take effect immediately, this assert may fail randomly in the future.

Reply:
Yes, the failure is exactly what I want to catch. Then we will have every reason to delay and check with a while loop.